sys/include/net/ng_pktbuf.h

```diff
@@ -59,68 +59,58 @@ extern "C" {
+ * @warning **Do not** change the ng_pktsnip_t::data and ng_pktsnip_t::size
  *          of a ng_pktsnip_t created by this function externally, except if
  *          they both are null or data is not from inside the packet buffer.
  *          This will most likely create memory leaks.
  *
- * @param[in,out] pkt   The packet you want to add a ng_pktsnip_t to. Leave
- *                      NULL if you want to create a new packet. Members may
- *                      change values; see above.
+ * @param[in] next      Next ng_pktsnip_t in the packet. Leave NULL if you
+ *                      want to create a new packet.
  * @param[in] data      Data of the new ng_pktsnip_t. If @p data is NULL no data
  *                      will be inserted into `result`. If @p data is already
  *                      in the packet buffer (e.g. a payload of an already
  *                      allocated packet) it will not be duplicated.
- * @param[in] size      Length of @p data. If @p size is 0 no data will be inserted
- *                      into the the packet buffer and ng_pktsnip_t::data will be
- *                      set to @p data.
+ * @param[in] size      Length of @p data. May not be 0.
  * @param[in] type      Protocol type of the ng_pktsnip_t.
  *
  * @return  Pointer to the packet part that represents the new ng_pktsnip_t.
  * @return  NULL, if no space is left in the packet buffer.
  * @return  NULL, if @p pkt != NULL, @p data = `pkt->data`,
  *                and @p size > `pkt->data`.
+ * @return  NULL, if @p size == 0.
  */
-ng_pktsnip_t *ng_pktbuf_add(ng_pktsnip_t *pkt, void *data, size_t size,
+ng_pktsnip_t *ng_pktbuf_add(ng_pktsnip_t *next, void *data, size_t size,
                             ng_nettype_t type);
 
 /**
+ * @brief   Marks the first @p size bytes in a received packet with a new
+ *          packet snip that is appended to the packet.
+ *
+ *  Graphically this can be represented as follows:
+ *  @code
+ * Before                                    After
+ * ======                                    =====
+ *                                                       (next)
+ *  pkt->data                                 result->data <== pkt->data
+ *  v                                         v                v
+ * +--------------------------------+        +----------------+---------------+
+ * +--------------------------------+        +----------------+---------------+
+ *  \__________pkt->size___________/          \_result->size_/ \__pkt->size__/
+ *  @endcode
+ *
+ * @pre @p pkt != NULL && @p size != 0
+ *
+ * @param[in] pkt   A received packets.
+ * @param[in] size  The size for the new packet snip.
+ * @param[in] type  The type of the new packet snip.
+ *
+ * @return  0 on success.
+ * @return  EINVAL, if pkt == NULL or size == 0.
+ * @return  ENOMEM, if no space is left in the packet buffer.
+ */
+int ng_pktbuf_mark(ng_pktsnip_t *pkt, size_t size, ng_nettype_t type);
```

Comment:
return value would need to be of type 'ng_pktsnip_t', right? Otherwise how would you get the `result` pointer?